### PR TITLE
Do not override value for composite types address and links when entering input

### LINKED ISSUE #6434: Recurring error "Record field type not supported : RELATION}"

### PR DESCRIPTION
**Pull request by ijreilly**
Closes #6434.

We don't want to override the values of the records' address or links as they are composite field and it is costly to loose the data. 
We will need a more unified behaviour here - maybe introduce a Ctrl+Z option.

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-field/hooks/useInitDraftValueV2.ts [hunk 1]
```diff
@@ -37,7 +38,10 @@ export const useInitDraftValueV2 = <FieldValue>() => {
           )
           .getValue();
 
-        if (isUndefined(value)) {
+        if (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/hooks/useInitDraftValueV2.ts:41]: why do we have this logic twice? :)

**Review by charlesBochet — COMMENTED** [review]
LGTM but curious why we have existing duplicated logic!

File: packages/twenty-front/src/modules/object-record/record-field/hooks/useInitDraftValueV2.ts (hunk 1 above)
Comment (ijreilly) [reply packages/twenty-front/src/modules/object-record/record-field/hooks/useInitDraftValueV2.ts:41]: right - I think we only want to use useInitDraftValueV2, so I deprecated the usage of initDraftValue from the other hook !

Comment (ijreilly) [reply packages/twenty-front/src/modules/object-record/record-field/hooks/useInitDraftValueV2.ts:41]: (cf https://github.com/twentyhq/twenty/pull/5023)

**Review by charlesBochet — APPROVED** [review]
LGTM!

**Pull request merged**